doc/STYLE.md

```diff
@@ -2,6 +2,27 @@
 
 ## Philosophy
 1. If it's inserting something into the existing classes/functions, try to follow the existing style as closely as possible.
-1. If it's brand new code or refactoring a complete class or area of the code, please follow as Modern C++ of a style as you can and reference the [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines) as much as you possibly can.
-1. When working with any Win32 or NT API, please try to use the [Windows Implementation Library](./WIL.md) smart pointers and result handlers.
-1. The use of NTSTATUS as a result code is discouraged, HRESULT or exceptions are preferred. Functions should not return a status code if they would always return a successful status code. Any function that returns a status code should be marked `noexcept` and have the `nodiscard` attribute.
+2. If it's brand new code or refactoring a complete class or area of the code, please follow as Modern C++ of a style as you can and reference the [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines) as much as you possibly can.
+3. When working with any Win32 or NT API, please try to use the [Windows Implementation Library](./WIL.md) smart pointers and result handlers.
+4. The use of NTSTATUS as a result code is discouraged, HRESULT or exceptions are preferred. Functions should not return a status code if they would always return a successful status code. Any function that returns a status code should be marked `noexcept` and have the `nodiscard` attribute.
+
+### Indentation
+
+Code and nested text should use multiples of 4 spaces of indentation, and no
+tab characters, like so:
+
+    void func(const int x)
+    {
+        std::cout << x << '\n';
+    }
+
+### Code Blocks
+
+Please use 4-space indentation to trigger code parsing, rather than [fenced code blocks](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks) or any other style, like so:

```

Comment:
the url here does not lead to the correct place for me. I also believe that we use fenced code blocks in our markdown code snippets so I'm not sure they should be discouraged.

Reply:
This was actully taken from the https://github.com/isocpp/CppCoreGuidelines, which was meant to be followed by https://github.com/Microsoft/Terminal/blob/master/doc/STYLE.md

Reply:
Oh, see I found this:

> consistently using indented vs. fenced is under discussion

and that more just sounds like a preference they've set for themselves, not one that needs to be enforced by other c++ projects. I actually quite prefer code fences, especially with language hinting at the top. Considering that's the style we've consistently used throughout the rest of our documentation, I'm inclined to keep it that way.

Reply:
I also like fenced code blocks. I don't think this needs to be strictly specified as they're both appropriate in their own contexts.